convert2rhel/checks.py

```diff
@@ -75,77 +89,63 @@ def perform_pre_ponr_checks():
 
 
 def check_convert2rhel_latest():
-    """Make sure that we are running the latest version of convert2rhel"""
-    gpg_path = os.path.join(utils.DATA_DIR, "gpg-keys", "RPM-GPG-KEY-redhat-release")
-    ssl_cert_path = os.path.join(utils.DATA_DIR, "redhat-uep.pem")
-    repo_content = (
-        "[convert2rhel]\n"
-        "name=Convert2RHEL Repository\n"
-        "baseurl=https://cdn.redhat.com/content/public/convert2rhel/$releasever/x86_64/os/\n"
-        "gpgcheck=1\n"
-        "enabled=1\n"
-        "sslcacert=%s\n"
-        "gpgkey=file://%s\n" % (ssl_cert_path, gpg_path)
-    )
+    """Make sure that we are running the latest downstream version of convert2rhel"""
+    logger.task("Prepare: Checking if this is the latest version of convert2rhel")
 
     cmd = [
         "repoquery",
         "--releasever=%s" % system_info.version.major,
     ]
 
-    repo_dir = tempfile.mkdtemp(prefix="convert2rhel_repo.", dir=utils.TMP_DIR)
-    utils.mkdir_p(repo_dir)
-    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
-    store_content_to_file(filename=repo_path, content=repo_content)
-
-    cmd.append("--setopt=reposdir=%s" % repo_dir)
+    cmd.append(
+        "--repofrom='convert2rhel,https://cdn.redhat.com/content/public/convert2rhel/$releasever/$basearch/os/'convert2rhel"
+    )
     cmd.append("convert2rhel")
 
     raw_output_convert2rhel_versions, return_code = run_subprocess(cmd, print_output=False)
 
-    if return_code != 0:
-        logger.warning(
-            "Couldn't check if this is the latest version of convert2rhel\n"
-            "repoquery failed %s, %s" % (return_code, raw_output_convert2rhel_versions)
-        )
-        return
-
     PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
     convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
-    latest_version = ("0", "0.00", "0")
+    latest_available_version = ("0", "0.00", "0")
 
+    # This loop will determine the latest available convert2rhel version in the yum repo.
+    # It assigns the epoch, version, and release ex: ("0", "0.26", "1.el7") to the latest_available_version variable.
     for package_version in convert2rhel_versions:
-        # rpm.lableCompare(pkg1, pkg2) compare two kernel version strings and return
-        # -1 if str2 is greater then str1, 0 if they are equal, 1 if str1 is greater the str2
-        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+        # rpm.lableCompare(pkg1, pkg2) compare two package version strings and return
+        # -1 if lastest_version is greater than package_version, 0 if they are equal, 1 if package_version is greater than lastest_version
+        ver_compare = rpm.labelCompare(package_version[1:], latest_available_version)
         if ver_compare > 0:
-            latest_version = package_version[1:]
+            latest_available_version = package_version[1:]
```

Comment:
```suggestion
            logger.debug("...convert2rhel version %s is newer than %s, updating latest known version" % (package_version[1:][1], latest_available_version[1]))
            latest_available_version = package_version[1:]
```

Reply:
I am a bit confused by this.
AD1 Do we need to inform the user about `updating latest known version`? If so, what is the purpose of this? Could the message be a bit clearer?
AD2 The log file displays this DEBUG message `...convert2rhel version 1.0 is newer than 0.00, updating latest known version`, even though the set version is supposed to be `42.0`, am I missing something @SpyTec @Andrew-ang9 ?
Thank you.

Reply:
hmmm I'm not 100% sure why it is showing 1.0 I will look into that and see what I can find. Thank you for catching this!

Reply:
1.0 is fine @Andrew-ang9, the 0.0 is the part, that confuses me

Reply:
ahhh my apologies, I believe that I have fixed this issue before I will try and go back and see if I can remember what I did with this!

Reply:
Hey @danmyway I think I see whats going on, I had changed the duplicated use of the cdn url and from what I am seeing I messed up the use of the format string to use it as one var. I am working on fixing it now and will push the fix as soon as I get it back working!

Thank you again for finding this!